src/app/services/captcha/captcha.factory.ts

```diff
@@ -0,0 +1,62 @@
+import { celebrate, Joi, Segments } from 'celebrate'
+import { RequestHandler } from 'express'
+import { okAsync, ResultAsync } from 'neverthrow'
+
+import FeatureManager, {
+  FeatureNames,
+  RegisteredFeature,
+} from '../../../config/feature-manager'
+import { MissingFeatureError } from '../../modules/core/core.errors'
+
+import {
+  CaptchaConnectionError,
+  MissingCaptchaError,
+  VerifyCaptchaError,
+} from './captcha.errors'
+import { CaptchaService } from './captcha.service'
+
+interface ICaptchaFactory {
+  verifyCaptchaResponse: (
+    response: string | null | undefined,
+    remoteip: string | undefined,
+  ) => ResultAsync<
+    true,
+    | CaptchaConnectionError
+    | VerifyCaptchaError
+    | MissingCaptchaError
+    | MissingFeatureError
+  >
+  validateCaptchaParams: RequestHandler
+}
+
+export const createCaptchaFactory = ({
+  isEnabled,
+  props,
+}: RegisteredFeature<FeatureNames.Captcha>): ICaptchaFactory => {
+  // Feature is enabled and valid.
+  if (isEnabled && props?.captchaPrivateKey) {
+    const captchaService = new CaptchaService(props.captchaPrivateKey)
+    return {
+      // Need to bind so this keyword works properly
+      verifyCaptchaResponse: captchaService.verifyCaptchaResponse.bind(
```

Comment:
question: is there a way to not use the `this` keyword then? bind seems like it will cause confusion. just curious, don't have to change it if this is the best way